ui-v2/app/styles/components/action-group/skin.scss

```diff
@@ -0,0 +1,40 @@
+.action-group {
+  @extend %action-group;
+}
```

Comment:
This is the same as `import JohnsActionGroup from 'action-group'`.

It means the selector you use for action-groups is decoupled from the actual action-group. You control the selector from the outside. The same as `require` and `import` let you control the name of the things you are importing/requiring from the outside.

In this instance its been left in the skin file by accident. But you'll also see it here:

https://github.com/hashicorp/consul/blob/feature/ui-css-refactor/ui-v2/app/styles/components/action-group.scss

Will remove from the skin file 👍

Reply:
`import` and `require` aren't perfect analogies because here you are setting the name/convention for the entire codebase. 

Say Consul does `.action-group { @extend %action-group; }` and Nomad does `.actions { @extend %action-group; }` and Vault does `.buttons { @extend %action-group; }`. 

How do have common documentation for this shared code? The dev experience for someone in the Vault codebase is to see `class="buttons"` somewhere and ask, "how does this work, what modifiers are available, how is this supposed to be used". Then, assuming there is shared design system documentation, the page they would want to see is called `%action-group`, not `.buttons`. And to get there, the dev would have to trace `.buttons` back through the Vault-specific CSS to get the imported placeholder name to know what to look for in the documentation.

It seems the benefits of sharing placeholders and not classes are

  1. Class names aren't prescriptive (or even required since any selector could extend a placeholder)
  2. Multiple selectors can extend the same placeholder
  3. Unused placeholders don't make it into the compiled css

I don't think Benefit 1 is actually a benefit. Benefit 2 sounds great on paper, but I don't see it being useful in practice. Benefit 3 is nice, but it can also be achieved via dead-code elimination from tools like [CSS Blocks](https://css-blocks.com/) or the underlying [OptiCSS](https://github.com/linkedin/opticss) tool.